### PR TITLE
feat(dbss): the DBSS instance resource supports new API

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

The DBSS instance resource supports new API.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/dbss" TESTARGS="-run TestAccInstance_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dbss -v -run TestAccInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccInstance_basic
=== PAUSE TestAccInstance_basic
=== CONT  TestAccInstance_basic
--- PASS: TestAccInstance_basic (1751.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dbss      1751.616s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/resources/dbss_instance.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ The following arguments are supported:
 * `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
   If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
 
-* `name` - (Required, String, ForceNew) Specifies the instance name. The name can contain `1` to `64` characters.
+* `name` - (Required, String) Specifies the instance name. The name can contain `1` to `64` characters.
   Only letters, digits, underscores (_), and hyphens (-) are allowed.
 
   Changing this parameter will create a new resource.
```
Comment (jinyangyang222) [inline docs/resources/dbss_instance.md:48]:
```suggestion
```

File: docs/resources/dbss_instance.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ The following arguments are supported:
...
 
-* `description` - (Optional, String, ForceNew) Specifies the description of the instance.
+* `description` - (Optional, String) Specifies the description of the instance.
 
   Changing this parameter will create a new resource.
```
Comment (jinyangyang222) [inline docs/resources/dbss_instance.md:116]:
```suggestion
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go [hunk 3]
```diff
@@ -610,6 +628,14 @@ func resourceInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta in
 	return resourceInstanceRead(ctx, d, meta)
 }
 
+func buildUpdateBodyParams(d *schema.ResourceData) map[string]interface{} {
```
Comment (jinyangyang222) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go:631]:
```suggestion
func buildUpdateNameAndDescriptionParams(d *schema.ResourceData) map[string]interface{} {
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go [hunk 4]
```diff
@@ -558,6 +555,37 @@ func resourceInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta in
...
+
+		respBody, err := utils.FlattenResponse(resp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		result := utils.PathSearch("result", respBody, "").(string)
+		if result != "success" {
+			return diag.Errorf("failed to update the DBSS instance")
+		}
```
Comment (jinyangyang222) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go:586]: The logic for updating the name and description fields can be extracted separately as a method, similar to updating the security group ID

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go [hunk 5]
```diff
@@ -143,12 +143,9 @@ func ResourceInstance() *schema.Resource {
 					"true", "false",
 				}, false),
 			},
-
 			"description": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Computed:    true,
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go:150]: Keep the Computed here. Deleting it will result in incompatible changes.

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dbss" TESTARGS="-run TestAccInstance_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dbss -v -run TestAccInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccInstance_basic
=== PAUSE TestAccInstance_basic
=== CONT  TestAccInstance_basic
--- PASS: TestAccInstance_basic (2007.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dbss      2007.046s
```

**Pull request merged**